### PR TITLE
(Closes #2624) Add `FortranWriter().schedule_node()`.

### LINKED ISSUE #2624: Schedule.debug_string() defaults to str() and FortranWriter raises an error on it.
`Schedule.debug_string()` defaults to `Node.debug_string()` which uses `DebugWriter` and defaults to `str()`.
`FortranWriter()(schedule)` raises a `VisitorError`.

Script:
```python

from psyclone.psyir.frontend.fortran import FortranReader
from psyclone.psyir.backend.fortran import FortranWriter
from psyclone.psyir.nodes import Schedule

reader = FortranReader()
writer = FortranWriter()

assignment_1 = "a = 1"
assignment_2 = "b = 2"

schedule = Schedule()
schedule.addchild(reader.psyir_from_statement(assignment_1))
schedule.addchild(reader.psyir_from_statement(assignment_2))
assert isinstance(schedule, Schedule)

print("Debug string of the Schedule:")
print(schedule.debug_string())

print("Fortran writer output:")
print(writer(schedule))
```

Output:
```bash
Debug string of the Schedule:
< Schedule:
Assignment[]
Reference[name:'a']Literal[value:'1', Scalar<INTEGER, UNDEFINED>]
Assignment[]
Reference[name:'b']Literal[value:'2', Scalar<INTEGER, UNDEFINED>]
End Schedule >

Fortran writer output:
Traceback (most recent call last):
  File "/home/julien/schedule_debug_string.py", line 20, in <module>
    print(writer(schedule))
          ^^^^^^^^^^^^^^^^
  File "/home/julien/miniconda3/lib/python3.12/site-packages/psyclone/psyir/backend/visitor.py", line 199, in __call__
    return self._visit(lowered_node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/julien/miniconda3/lib/python3.12/site-packages/psyclone/psyir/backend/visitor.py", line 304, in _visit
    raise VisitorError(
psyclone.psyir.backend.visitor.VisitorError: Visitor Error: Unsupported node 'Schedule' found: method names attempted were ['schedule_node', 'scopingnode_node', 'node_node'].
```

In my opinion both should output:
```fortran
a = 1
b = 2
```

### PR DESCRIPTION
**Pull request by JulienRemy**
Avoids defaulting to `str(schedule)` for `Schedule().debug_string()` and raising a `VisitorError` using `FortranWriter()(Schedule())`.

### CODE REVIEW

File: src/psyclone/psyir/nodes/schedule.py
Comment (sergisiso) [inline src/psyclone/psyir/nodes/schedule.py:104]: I don't think we need this (and associated test) now that a "schedule_node" visitor has been introduced to the Writer, do we?

Comment (JulienRemy) [reply src/psyclone/psyir/nodes/schedule.py]: You're right, I'd forgotten this defaults to `DebugWriter` and thus to `FortranWriter`. They're now removed.

**Comment by JulienRemy** [comment]
@sergisiso This is a very minor feature but it's proven convenient at times. It should now be ready to merge if you want.

**Review by sergisiso — APPROVED** [review]
Thanks @JulienRemy, this is a small PR that improves our debug_string capabilities, all checks are green. This is approved for merging.

Apologies that this was not reviewer earlier, in the future remember to put the "Ready for Review" label so I can see it marked on the list of PRs.

**Comment by sergisiso** [comment]
Waiting for #2775 for get merged

**Pull request merged**
